configs/preferences_txt_serializer.go

```diff
@@ -1,147 +0,0 @@
-/*
- * This file is part of arduino-cli.
- *
- * Copyright 2018 ARDUINO SA (http://www.arduino.cc/)
- *
- * This software is released under the GNU General Public License version 3,
- * which covers the main part of arduino-cli.
- * The terms of this license can be found at:
- * https://www.gnu.org/licenses/gpl-3.0.en.html
- *
- * You can be released from the requirements of the above licenses by purchasing
- * a commercial license. Buying such a license is mandatory if you want to modify or
- * otherwise use the software for commercial activities involving the Arduino
- * software without disclosing the source code of your own applications. To purchase
- * a commercial license, send an email to license@arduino.cc.
- */
-
-package configs
-
-import (
-	"errors"
-	"net/url"
-	"os"
-	"strings"
-
-	"github.com/arduino/go-paths-helper"
-
-	"github.com/arduino/go-properties-orderedmap"
-	"github.com/sirupsen/logrus"
-)
-
-// IsBundledInDesktopIDE returns true if the CLI is bundled with the Arduino IDE.
-func (config *Configuration) IsBundledInDesktopIDE() bool {
-	if config.IDEBundledCheckResult != nil {
-		return *config.IDEBundledCheckResult
-	}
-
-	res := false
-	config.IDEBundledCheckResult = &res
-
-	logrus.Info("Checking if CLI is Bundled into the IDE")
-	executable, err := os.Executable()
-	if err != nil {
-		logrus.WithError(err).Warn("Cannot get executable path")
-		return false
-	}
-	executablePath := paths.New(executable)
-	if err := executablePath.FollowSymLink(); err != nil {
-		logrus.WithError(err).Warn("Cannot get executable path")
-		return false
-	}
-	ideDir := executablePath.Parent()
-	logrus.Info("Candidate IDE Directory: ", ideDir)
-
-	tests := []string{
-		"tools-builder",
-		"examples/01.Basics/Blink",
-	}
-	for _, test := range tests {
-		if !ideDir.Join(test).Exist() {
-			return false
-		}
-	}
-
-	portable := "portable"
-	if ideDir.Join(portable).Exist() {
-		logrus.Info("IDE is portable")
-		config.IsPortable = true
-	}
-
-	config.ArduinoIDEDirectory = ideDir
-	res = true
-	return true
-}
-
-// LoadFromDesktopIDEPreferences loads the config from the Desktop IDE preferences.txt file
-func (config *Configuration) LoadFromDesktopIDEPreferences() error {
```

Comment:
Are we dropping loading/overlaying options from the IDE `preferences.txt`? (just asking I may be OK with that :-))

Reply:
At first I didn't fully get what this was for - I'd say let's get rid of it. 
If needed, I'd add a specific command like `arduino-cli config import /path/to/preferences.txt` so the operation is explicit.